html/semantics/forms/the-form-element/form-elements-filter.html

```diff
@@ -169,4 +174,17 @@
 
   assert_array_equals(ids, allCorrectIDs);
 });
+
+test(() => {
+  const shadowRoot1 = document.querySelector("#shadow-1").attachShadow({mode: "open"});
+  const input1 = document.createElement("input");
+  shadowRoot1.appendChild(input1);
+
+  const shadowRoot2 = document.querySelector("#shadow-2").attachShadow({mode: "open"});
+  const input2 = document.createElement("input");
+  input2.setAttribute("form", "form2");
+  shadowRoot2.appendChild(input2);
+
+  assert_equals(document.querySelector("#form2").elements.length, 0);
```

Comment:
Was thinking about it, but wasn't sure if it belonged here or with tests for the `form` attribute. Added now. :)